### PR TITLE
Added post validation steps for NS loadbalancing scenarios

### PR DESCRIPTION
**Pull request by HaruChebrolu**
# Description

Added post validation steps for NS loadbalancing scenarios.
Logs: http://magna002.ceph.redhat.com/cephci-jenkins/harika/ns_loadbalancing/cephci-run-NLZQ8M/

<summary>click to expand checklist</summary>

- [x] Create a test case in Polarion reviewed and approved.
- [x] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [x] Review the automation design
- [x] Implement the test script and perform test runs
- [x] Submit PR for code review and approve
- [x] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/nvmeof/test_ceph_nvmeof_loadbalancing.py [hunk 1]
```diff
@@ -236,48 +249,79 @@ def run(ceph_cluster: Ceph, **kwargs) -> int:
...
-                                configure_listeners(ha, [scaleup_node], subsys_args)
+                        old_namespaces = parse_namespaces(config, old_namespaces)
+                        ha.scale_up(scaleup_nodes, gw_nodes, old_namespaces)
+
+                        # Validate IO for old namespaces
+                        ha.validate_scaleup(scaleup_nodes, old_namespaces)
+
+                        # Add listeners and namespaces to newly added GWs
+                        for subsys_args in config["subsystems"]:
+                            LOG.info(f"Adding listeners for {scaleup_nodes}")
```
Comment (rahullepakshi) [inline tests/nvmeof/test_ceph_nvmeof_loadbalancing.py:287]: @HaruChebrolu It is better to add all listeners first and then move to new namespace addition in interest of time and IO getting stuck . Though they are new GWs, it is always advised to add all listeners before adding namespaces

**Review by rahullepakshi — COMMENTED** [review]
gw info logs too many times, if possible lets try to avoid duplication and remove it if its part of GW initialization

File: tests/nvmeof/test_ceph_nvmeof_loadbalancing.py (hunk 1 above)
Comment (rahullepakshi) [reply tests/nvmeof/test_ceph_nvmeof_loadbalancing.py:287]: The order below would be more beneficial for scale up new Gws

1. Run IO on old namespaces 
2. ceph orch apply new GW nodes
3. Add listeners on all subsystems for alll new GW nodes immediately
4. Then add namespaces
5. Disconnect, discover and connect again from initiators

Currently it is 2,1,4-3,5

**Comment by HaruChebrolu** [comment]
> gw info logs too many times, if possible lets try to avoid duplication and remove it if its part of GW initialization

I will try to mitigate as much as I can, but I am little skeptical if I can completely avoid it.

File: tests/nvmeof/test_ceph_nvmeof_loadbalancing.py (hunk 1 above)
Comment (HaruChebrolu) [reply tests/nvmeof/test_ceph_nvmeof_loadbalancing.py:287]: Will update the loop to add listeners to all subsystems and then move to adding namespaces.
And currently I running IO on old namespaces and then I am applying new GW nodes, please let me know if I am missing something.

Comment (rahullepakshi) [reply tests/nvmeof/test_ceph_nvmeof_loadbalancing.py:287]: Thanks @HaruChebrolu 
I got confused with https://github.com/red-hat-storage/cephci/blob/a16e3a89f636bd905fc9900e35973122b9e48a54/tests/nvmeof/test_ceph_nvmeof_loadbalancing.py#L280-L283

i think you are validating IO for old namespaces after orch apply new nodes? If its case then its fine. thanks
Another suggestionis you can validate IO for old namespaces as well as new namespaces together after listener add and namespace add. We can do this way too

Comment (HaruChebrolu) [reply tests/nvmeof/test_ceph_nvmeof_loadbalancing.py]: Done

**Review by rahullepakshi — APPROVED** [review]

**Review by Manohar-Murthy — APPROVED** [review]
lgtm

**Pull request merged**